model/llm/llm.go

```diff
@@ -99,14 +103,30 @@ func (m *Model) GenerateTestsForFile(logger *log.Logger, language language.Langu
 		return nil, err
 	}
 
-	logger.Printf("Querying model %q with:\n%s", m.ID(), string(bytesutil.PrefixLines([]byte(request), []byte("\t"))))
-	start := time.Now()
-	response, err := m.provider.Query(context.Background(), m.model, request)
-	if err != nil {
+	var response string
+	var duration time.Duration
+	if err := retry.Do(
+		func() error {
+			logger.Printf("Querying model %q with:\n%s", m.ID(), string(bytesutil.PrefixLines([]byte(request), []byte("\t"))))
+			start := time.Now()
+			response, err = m.provider.Query(context.Background(), m.model, request)
+			if err != nil {
+				return err
+			}
+			duration = time.Since(start)
+			logger.Printf("Model %q responded (%d ms) with:\n%s", m.ID(), duration.Milliseconds(), string(bytesutil.PrefixLines([]byte(response), []byte("\t"))))
+
+			return nil
+		},
+		retry.Delay(2*time.Second),
+		retry.LastErrorOnly(true),
+		retry.Attempts(m.retries),
+		retry.RetryIf(func(err error) bool {
+			return true
+		}),
+	); err != nil {
```

Comment:
Please add a [`retry.OnRetry`](https://pkg.go.dev/github.com/avast/retry-go/v4#OnRetry) function that logs the retries and the error. The `retry` docs have a nice example.

Watch out, this will change the test case where you assume that the error is never in the logs.